### DISCUSSION TITLE
Filtering document library by List Column?

### DISCUSSION THREAD

**Discussion by RabidDolphin** [O365/python-o365#1000] [Q&A]
I am trying to use new_query to filter over a document library by using list columns. Not sure I am understanding queries. 

Here is an example of what I have tried: 

```
drive = site.get_document_library(drive_id) 
ls = site.get_list_by_name(drive.name)                  # grabs drive as a list using name 
col = ls.get_list_columns()[7]                                 #  gets the desired list column

query = drive.new_query().on_list_field(col.display_name).equal("Desired List Value Here")

drive.get_items(query = query)
```

The last line returns a 400 Client Error: Bad Request for url.
Additionally, after using a query, calling `drive.get_items()` doesn't work as usual - it only performs as if I passed the query. 

Can anyone offer any advice on if I'm using queries incorrectly? Thank you!!

**Comment by alejcas** [comment]
I think the OData query is not correct. Check the query (`print(query)`) to see what the code constructed and see if this query works in ms graph explorer.
